### PR TITLE
Create keycodes for RGB control functions

### PR DESCRIPTION
**Pull request by TerryMathews**
Moves RGB controls out of the macro function and assigns them their own
keycodes:
RGB_TOG (toggle on/off)
RGB_MOD (mode step)
RGB_HUI (increase hue)
RGB_HUD (decrease hue)
RGB_SAI (increase saturation)
RGB_SAD (decrease saturation)
RGB_VAI (increase brightness)
RGB_VAD (decrease brightness)

### CODE REVIEW

**Comment by TerryMathews** [comment]
I have been testing this updated code on my Phantom for at least 30 minutes while I cleaned up the other keyboard trees. All the RGB functions work as expected and everything appears to work as before.

File: quantum/quantum.c [hunk 1]
```diff
@@ -103,6 +103,56 @@ bool process_record_quantum(keyrecord_t *record) {
           return false;
       }
       break;
+	#ifdef RGBLIGHT_ENABLE
+	case RGB_TOG:
+		if (record->event.pressed) {
+			rgblight_toggle();
+			return false;
```
Comment (jackhumbert) [inline quantum/quantum.c]: These `return false;` statements should really go outside the if block - we don't want them to be processed (what returning false does) either way.

Comment (TerryMathews) [inline quantum/quantum.c]: So then should it look like:

`}`
`return false;`
`break;`
?

And if so, do you want me to fix RESET and DEBUG? They have the same mistake - they were what I used as examples for my code to look like.

Comment (jackhumbert) [inline quantum/quantum.c]: Yeap! Ah, right on - sure, if you wouldn't mind!

**Comment by TerryMathews** [comment]
Moving return false; like that broke RGB controls on my Phantom and is making the CAPS and SCROLL LEDs very dim.

**Comment by TerryMathews** [comment]
I work on this some more tomorrow. Evidently, I had my teensy flasher pointed at a different repository and didn't notice that the LED brightness had changed. I've got the change narrowed down to between #407 and #430. #407 is normal brightness.

Since the changes I made had nothing to do with that regression, I will finish this pull request then treat the LED brightness as a separate issue.

**Comment by TerryMathews** [comment]
Code in quantum.c has been updated as we discussed. Fix for Phantom LED brightness discussed in issue #495 got rolled up into this PR as well since it was a *very* simple fix. This is compiling and working as expected on my Phantom.

After Travis finishes checking the code, I believe this is ready to merge.

**Comment by jackhumbert** [comment]
Awesome! Thanks - sorry about the LED stuff. That was probably my bad when converting that stuff over :)

**Comment by TerryMathews** [comment]
No worries, just glad it's fixed.

**Pull request merged**